internal/controller/promotionstrategy_controller.go

```diff
@@ -133,60 +133,44 @@ func (r *PromotionStrategyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	logger := log.FromContext(ctx)
 
-	pc := promoterv1alpha1.ProposedCommit{}
 	pcName := utils.KubeSafeUniqueName(ctx, utils.GetProposedCommitName(ps.Name, environment.Branch))
-	err := r.Get(ctx, client.ObjectKey{Namespace: ps.Namespace, Name: pcName}, &pc, &client.GetOptions{})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			logger.Info("ProposedCommit not found, creating", "namespace", ps.Namespace, "name", pcName)
-
-			// The code below sets the ownership for the Release Object
-			kind := reflect.TypeOf(promoterv1alpha1.PromotionStrategy{}).Name()
-			gvk := promoterv1alpha1.GroupVersion.WithKind(kind)
-			controllerRef := metav1.NewControllerRef(ps, gvk)
 
-			pc = promoterv1alpha1.ProposedCommit{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:            pcName,
-					Namespace:       ps.Namespace,
-					OwnerReferences: []metav1.OwnerReference{*controllerRef},
-					Labels: map[string]string{
-						"promoter.argoproj.io/promotion-strategy": utils.KubeSafeLabel(ctx, ps.Name),
-						"promoter.argoproj.io/proposed-commit":    utils.KubeSafeLabel(ctx, pcName),
-						"promoter.argoproj.io/environment":        utils.KubeSafeLabel(ctx, environment.Branch),
-					},
-				},
-				Spec: promoterv1alpha1.ProposedCommitSpec{
-					RepositoryReference:    ps.Spec.RepositoryReference,
-					ProposedBranch:         fmt.Sprintf("%s-%s", environment.Branch, "next"),
-					ActiveBranch:           environment.Branch,
-					ActiveCommitStatuses:   append(environment.ActiveCommitStatuses, ps.Spec.ActiveCommitStatuses...),
-					ProposedCommitStatuses: append(environment.ProposedCommitStatuses, ps.Spec.ProposedCommitStatuses...),
-				},
-			}
+	// The code below sets the ownership for the Release Object
+	kind := reflect.TypeOf(promoterv1alpha1.PromotionStrategy{}).Name()
+	gvk := promoterv1alpha1.GroupVersion.WithKind(kind)
+	controllerRef := metav1.NewControllerRef(ps, gvk)
+
+	pc := promoterv1alpha1.ProposedCommit{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            pcName,
+			Namespace:       ps.Namespace,
+			OwnerReferences: []metav1.OwnerReference{*controllerRef},
+			Labels: map[string]string{
+				"promoter.argoproj.io/promotion-strategy": utils.KubeSafeLabel(ctx, ps.Name),
+				"promoter.argoproj.io/proposed-commit":    utils.KubeSafeLabel(ctx, pcName),
+				"promoter.argoproj.io/environment":        utils.KubeSafeLabel(ctx, environment.Branch),
+			},
+		},
+		Spec: promoterv1alpha1.ProposedCommitSpec{
+			RepositoryReference:    ps.Spec.RepositoryReference,
+			ProposedBranch:         fmt.Sprintf("%s-%s", environment.Branch, "next"),
+			ActiveBranch:           environment.Branch,
+			ActiveCommitStatuses:   append(environment.ActiveCommitStatuses, ps.Spec.ActiveCommitStatuses...),
+			ProposedCommitStatuses: append(environment.ProposedCommitStatuses, ps.Spec.ProposedCommitStatuses...),
+		},
+	}
 
+	err := r.Patch(ctx, &pc, client.MergeFrom(&promoterv1alpha1.ProposedCommit{}))
```

Comment:
Why patch rather than update? Since we own the resource, update seems safer.

Reply:
To avoid conflicts with the proposed commit controllers status updates etc